mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICalendarParser.java

```diff
@@ -134,10 +136,28 @@ public String getMailetInfo() {
             ByteArrayInputStream inputStream = new ByteArrayInputStream(icsContent);
             return Stream.of(Pair.of(key, builder.build(inputStream)));
         } catch (IOException e) {
-            LOGGER.error("Error while reading input: " + icsContent, e);
+            if (LOGGER.isErrorEnabled()) {
+                String icsContentAsString;
+                try {
+                    icsContentAsString = new String(icsContent, Charset.forName("UTF-8"));
+                } catch (Throwable t) {
+                    LOGGER.error("Error while decoding ics content", t);
+                    icsContentAsString = new String(Hex.encodeHex(icsContent));
+                }
+                LOGGER.error("Error while reading input: " + icsContentAsString, e);
+            }
             return Stream.of();
         } catch (ParserException e) {
-            LOGGER.error("Error while parsing ICal object: " + icsContent, e);
+            if (LOGGER.isErrorEnabled()) {
+                String icsContentAsString;
```

Comment:
You have code duplication. Please prefer extracting this byte array -> string method.

Reply:
Done